tests/libraries/test_reentrancy_guard.py

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
+
+from ethereum.tools import tester
+from ethereum.tools.tester import TransactionFailed
+from pytest import fixture, mark, lazy_fixture, raises
+
+
+@fixture(scope='session')
+def testerSnapshot(sessionFixture):
+    sessionFixture.resetSnapshot()
+    sessionFixture.uploadAndAddToController('solidity_test_helpers/ReentrancyGuardHelper.sol')
+    ReentrancyGuardHelper = sessionFixture.contracts['ReentrancyGuardHelper']
+    return sessionFixture.chain.snapshot()
+
+
+@fixture
+def testerContractsFixture(sessionFixture, testerSnapshot):
+    sessionFixture.chain.revert(testerSnapshot)
```

Comment:
There are several other things that need to happen along with reverting the snaptshot.  The following is what happens in `sessionFixture.resetSnapshot()`.
```
        self.chain.block = self.originalBlock
        self.chain.revert(self.snapshot)
        self.chain.head_state = self.originalHead
        self.contracts = deepcopy(self.originalContracts)
```

Also, we do the following when creating snapshots, which we should also do:
```
        self.originalHead = self.chain.head_state
        self.originalBlock = self.chain.block
        self.originalContracts = deepcopy(self.contracts)
        self.snapshot = self.chain.snapshot()
```

It definitely seems like there would be value in having the `ContractsFixture` have a method for create snapshot that returns an object containing all of those pieces of data and another method for resetting a snapshot that takes in the same object type.  This way we don't have to keep re-implementing everything correctly anytime we want a new derived fixture.